File: src/frogger/screen/frame/elements/frog/Frog.java
package frogger.screen.frame.elements.frog;

import javafx.scene.Node;


public class Frog {

	private Node frog;

	public Frog(Node frog){
		this.frog = frog;
	}

	public Node getFrog(){
		return frog;
	}
	public void moveFrogBy(int dx, int dy) {
		PositionCalculator.moveFrogBy(frog, dx, dy);
	}

	public void moveFrogTo(double x, double y) {
		PositionCalculator.moveFrogTo(frog, x, y);
	}
}

File: src/frogger/screen/frame/elements/frog/Frog.scala
package frogger.screen.frame.elements.frog

import frogger.screen.frame.helpers.PositionCalculator
import javafx.scene.Node

class Frog(var frog: Node) {

  val positionCalculator = new PositionCalculator

  def getFrog(): Node ={
    if(frog == null) throw new Exception("Erro no jogo, tente jogar novamente!")
    return frog
  }
  def moveFrog(dx: Int, dy: Int): Boolean = {
    val boundsInScene = frog.localToScene(frog.getBoundsInLocal)
    if (boundsInScene.getMinY > 10) {
      positionCalculator.moveFrog(frog, dx, dy)
      return true
    }
    return false
  }

  def moveFrog(x: Double, y: Double): Unit = {
    positionCalculator.moveFrog(frog, x, y)
  }

  /*def switchFrog(keyCode: KeyCode): Unit = {
    PositionCalculator.switchPosition(keyCode)
  }*/

  /*def setLastKeyPressedToFalse(): Unit = {
    PositionCalculator.setFalseToLastKeyActive()
  }*/

  /*def switchFrogPositionAndImage(keyCode: KeyCode): Unit = {
    PositionCalculator.switchDirectionAndImage(keyCode)
  }*/

}
